rbac/management/principal/it_service.py

```diff
@@ -275,7 +280,36 @@ def get_service_accounts_group(self, group: Group, bearer_token: str, options: d
             service_account_principals=sap_dict, it_service_accounts=it_service_accounts, options=options
         )
 
-        return service_accounts
+        # If either the description or name filters were specified, we need to only keep the service accounts that
+        # match that criteria.
+        sa_description_filter = options.get(SERVICE_ACCOUNT_DESCRIPTION_KEY)
+        sa_name_filter = options.get(SERVICE_ACCOUNT_NAME_KEY)
+
+        if sa_description_filter or sa_name_filter:
+            filtered_service_accounts: list[dict] = []
+
+            for sa in service_accounts:
+                # Initialize both matches as "True", because having them at "False" would force us to make them "True"
+                # whenever one of the filters is not specified, since otherwise not specifying a filter would stop the
+                # service account to be included in the resulting list.
+                sa_description_filter_matches: bool = True
+                sa_name_filter_matches: bool = True
+
+                if sa_description_filter:
+                    sa_description_filter_matches = ("description" in sa) and (
+                        sa_description_filter in sa["description"]
```

Comment:
I have echoed the question in the ticket and mentioned you too, to see what the rest think about it, because it is a interesting point. Thanks for the review!

Reply:
I am going to resolve this conversation since it seems that we are going to make it case sensitive, as in the PR.